src/main/kotlin/com/openlattice/organizations/ExternalDatabaseManagementService.kt

```diff
@@ -0,0 +1,378 @@
+                                    privileges.addAll(listOf("INSERT", "UPDATE"))
+                                }
+                                if (ace.permissions.contains(Permission.READ)) {
+                                    privileges.add("SELECT")
+                                }
+                            }
+                            val grantSql = createPrivilegesSql(action, privileges, tableAndColumnNames.first, tableAndColumnNames.second, dbUser)
+                            stmt.addBatch(grantSql)
+                        }
+                    }
+                    stmt.executeBatch()
+                    dataSource.connection.commit()
+                }
+            }
+        }
+    }
+
+    /**
+     * Revokes all privileges for a user on an organization's database
+     * when that user is removed from an organization.
+     */
+    fun revokeAllPrivilegesFromMember(orgId: UUID, userId: String) {
+        val dbName = PostgresDatabases.buildOrganizationDatabaseName(orgId)
+        val userName = getDBUser(userId)
+        assemblerConnectionManager.connect(dbName).use {
+            it.connection.createStatement().use { stmt ->
+                stmt.execute("REVOKE ALL ON DATABASE $dbName FROM $userName")
+            }
+        }
+    }
+
+    fun addPermissions(dbName: String, orgId: UUID, tableId: UUID, tableName: String, maybeColumnId: Optional<UUID>, maybeColumnName: Optional<String>) {
+        val privilegesByUser = HashMap<String, MutableSet<String>>()
+        var columnCondition = ""
+        lateinit var aclKey: AclKey
+        if (maybeColumnId.isPresent && maybeColumnName.isPresent) {
+            val columnName = maybeColumnName.get()
+            val columnId = maybeColumnId.get()
+            columnCondition = "AND column_name = '$columnName'"
+            aclKey = AclKey(orgId, tableId, columnId)
+        } else {
+            aclKey = AclKey(orgId, tableId)
+        }
+        val sql = "SELECT grantee AS user, privilege_type " +
+                "FROM information_schema.role_table_grants " +
+                "WHERE table_name = '$tableName' " +
+                columnCondition
+        BasePostgresIterable(
+                StatementHolderSupplier(assemblerConnectionManager.connect(dbName), sql)
+        ) { rs ->
+            val user = user(rs)
+            val privilege = privilegeType(rs).toUpperCase()
+            privilegesByUser.getOrPut(user) { mutableSetOf() }.add(privilege)
+        }
+        privilegesByUser.forEach { (user, privileges) ->
+            val principal = securePrincipalsManager.getPrincipal(user).principal
+            val aceKey = AceKey(aclKey, principal)
+            var permissions = EnumSet.noneOf(Permission::class.java)
+            if (privileges == OwnerPrivileges.values() as Set<String>) { //eek
+                permissions.add(Permission.OWNER)
+            }
+            if (privileges.contains("SELECT")) {
```

Comment:
You could use `OwnerPrivileges` also here, but maybe with renaming them as PostgresTablePrivileges or something like that. Otherwise, if the only purpose of that enum class is to check, if the user has all of those privileges, I would move it to this class as a set of string.

Reply:
This makes so much more sense, I'll change that